src/main/java/teammates/common/datatransfer/questions/FeedbackContributionQuestionDetails.java

```diff
@@ -52,13 +53,15 @@ public FeedbackContributionQuestionDetails() {
 
     public FeedbackContributionQuestionDetails(String questionText) {
         super(FeedbackQuestionType.CONTRIB, questionText);
-        isNotSureAllowed = true;
+        isZeroSum = true;
+        isNotSureAllowed = false;
```

Comment:
It should be the other way around (`isZeroSum = false, isNotSureAllowed = true`), because those are the least restrictive functions. Unless we're going to change how this question type works in a backward-incompatible way.

Reply:
> This option can be selected by default. The instructor can be cautioned not to unselect it if marks are intended to be given based on ratings received.

Hm this was suggested by @damithc when the issue opened and I think it makes sense to set `isZeroSum` as `true` by default in this case. Your views?

Reply:
@damithc you're suggesting to change how the question type works in a backward-incompatible way. Are you 100% sure?

Reply:
> @damithc you're suggesting to change how the question type works in a backward-incompatible way. Are you 100% sure?

I see what you mean. The flag should be selected by default _in the UI_ when creating a new question i.e., applies to future only. However, the default value for existing question should be false (as this constraint was never applied to existing questions). Is that possible to achieve? Or do we need a data migration for this?

Reply:
It is possible, but backward-compatibility goes further than that. This is a destructive change to users who are already used to creating contribution questions.

Reply:
>It is possible, but backward-compatibility goes further than that. This is a destructive change to users who are already used to creating contribution questions.

Good point. Thinking further, I guess we can divide users into these categories:
1. Users who are aware of the current not-zero-sum approach, and prefer the current approach. If we make zero-sum as the default, these users might continue to create questions without realizing that the question is now zero-sum. This is not optimal. 
2. User who are unaware of the current not-zero-sum approach and misuse this question type (i.e., by attaching grades/marks to absolute contribution ratings) and get greatly annoyed later when they find out about this 'unexpected' behavior when students complain. This can be prevented by making the zero-sum option as the default.
3. Everyone else: not generally affected.

Given the current not-zero-sum behavior is not 'usual' doesn't always produce an optimal outcome (e.g., category 3), I think it shouldn't be the default. However, changing the default to zero-sum affects the category 1 above (unless they simply create new sessions by copying old sessions). Perhaps we can use the notifications feature to alert existing users that the default behavior of this question type has changed? What do you think?

Reply:
- 2 hardly counts. Users misusing this question type (or any other question type) doesn't change the fact that this is the service that we are offering.
- 1 on the other hand is not what any respectable production system should be doing (without telling).

Reply:
Yes, we shouldn't change without informing users (and shouldn't change at all if possible -- but I think this change is for the better). Let's go with making zero-sum the default for future team-contribution questions. We'll announce it to users (via the notification feature) at the same time the new version is released.